File: java-code/src/runnables/ReadFile.java
package runnables;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class ReadFile implements Runnable {

  private final String _fileName;

  public ReadFile(String fileName) {
    this._fileName = fileName;
  }

  @Override
  public void run() {
    try (BufferedReader reader = new BufferedReader(new FileReader(_fileName))) {
      String line;
      System.out.println("Reading file: " + this._fileName);
      System.out.println("==============================================================");
      while ((line = reader.readLine()) != null) {
        System.out.println(line);
      }
      System.out.println("==============================================================");
      System.out.println("End of file!");
    } catch (IOException e) {
      System.err.println("Error reading file: " + e.getMessage());
    }
  }
}

File: c#-code/src/runnables/ReadFile.cs
namespace APLCAssignment3.Runnables
{
    using System.IO;
    using System; 

    public class ReadFile
    {
        private readonly string _fileName;

        public ReadFile(string fileName)
        {
            this._fileName = fileName;
        }


        public void Run() 
        {
            try 
            {
                Console.WriteLine("Reading file: " + this._fileName);
                Console.WriteLine("==============================================================");

                foreach (string line in File.ReadLines(_fileName))
                {
                    Console.WriteLine(line);
                }

                Console.WriteLine("==============================================================");
                Console.WriteLine("End of file");
            }
            catch (IOException e)
            {
                Console.Error.WriteLine($"Unable to read file {e.Message}");
            }
        }
    }
}